### ISSUE TITLE
KeePassXC-Browser looses connection after closing Firefox 66.x.x

### ISSUE THREAD

**Issue by Dadanaut** [keepassxreboot/keepassxc-browser#455]
I can connect to my database once. But after I’ve closed Firefox and start it again the connectiong has got lost. I can reconnect again but it happens again and again...

I’ve read the troubleshooting guide but didn’t recognize something wrong.

We use KeePassXC at work too, and the problem happens there too. On 4 other macs.

What smells a bit fishy is that after I start Firefox there’s always a welcome message from one of the addons (HTTPS everywhere for example).

## Debug info

Console message:

Connecting to native messaging host org.keepassxc.keepassxc_browser keepass.js:999:5
Error: Content script already unregistered
ExtensionUtils.jsm
Server public key: GTivIAaGJRBf+YWPlzKqdBoDKLe89DwohN3t5AQ38R0= keepass.js:629:17
Error 11: No saved databases found.
keepass.js:1083:5
Error: Promised response from onMessage listener went out of scope init.js:81:1

​KeePassXC - 2.3.4
KeePassXC-Browser - 1.3.3
Operating system: MacIntel (macOS Sierra, 10.12.6)
Browser: Mozilla Firefox 66.0.2

**Comment by varjolintu** [comment]
Have you tried to correct the permissions? Maybe your profile directory is not working correctly and the localStorage is cleaned after every exit.

**Comment by Dadanaut** [comment]
Thanks for the hint. But that didn’t help either.

I did only a permission check and repair for my system globally.

**Comment by varjolintu** [comment]
Next steps:
1) Connect y.our database to KeePassXC
2) Go to extension's settings page (Connected databases) and see if the connected ID matches with the one saved to KeePassXC (found from Database settings).
3) Close KeePassXC and reopen it. Check step 2 again and make sure the ID has not disappeared from KeePassXC.
4) Close Firefox and reopen it. Does the Connected databases still show the saved ID? If not, inspect the page (Tools -> Web Developer -> Inspector) and see if there's any error messages in the console.

**Comment by Dadanaut** [comment]
The ids were the same till I’ve reopned Firefox again. Then the database connection was lost again.

The console says:

Source-Map-Fehler: TypeError: NetworkError when attempting to fetch resource.
Ressourcen-Adresse: moz-extension://2a796784-5a53-9a42-a65b-597779069c87/options/bootstrap.min.css
Source-Map-Adresse: bootstrap.min.css.map

**Comment by varjolintu** [comment]
Have you tried if KeePassXC 2.4.0 works any better?

**Comment by Dadanaut** [comment]
Damn! No change :-(

**Comment by Dadanaut** [comment]
It works wonderfully with Google Chrome.

Shall I start over with a clean Firefox installation?

**Comment by varjolintu** [comment]
You can try to make a new temporary profile to Firefox and see if everything works with it before cleaning your main profile or erasing other stuff.

**Comment by mygithuber** [comment]
Similar issue with FF-66.0.2 (64 on Win7-64):
If rebooting system and starting 1.KeePassXC and 2. FF with browser addon, connection KeePassXC-Browser v1.3.3 <=> KeePassXC v2.4.0 has been lost and must be reestablished once by typing a new connection key...

Above steps 1. > 4. from @varjolintu show no fault while running once booted Win7-system.

**Comment by Dadanaut** [comment]
No changes with a new profile or a fresh installation of Firefox.

**Comment by varjolintu** [comment]
And you are still having the first time welcome messages from one of the other addons?

**Comment by Dadanaut** [comment]
Yep.

**Comment by Dadanaut** [comment]
But only after the second start of Firefox ... after the first the connection between the database and Firefox is okay. And there’s no welcoming message of an add-on.

After the second restart of Firefox the connection got lost ... and the welcoming message appears.

**Comment by varjolintu** [comment]
Do you have some modifications in user.js, or does this problem occur if you disable your anti-virus software? I really doubt this have anything to do with the extension itself.

**Comment by Dadanaut** [comment]
Changes to user.js? Nope. Didn’t came to my mind until now ;-)

I’m not using any anti virus software.

And yes, I think that the problem is Firefox too.

I’ve installed 56.0.2 and it seems to work ... .. .

**Comment by mygithuber** [comment]
How to add Label 'WindowsOS' to this issue? 
It is not only present on platform 'macOS' but on 'WindowsOS' too...

KeePassXC-browser for Firefox forgets connection to 'Connected databases' as describben above:
https://github.com/keepassxreboot/keepassxc-browser/issues/455#issuecomment-478113623

**Comment by varjolintu** [comment]
@mygithuber You have to figure out why Firefox forgets the extension settings. It's not related to the extension itself.

**Comment by mygithuber** [comment]
Hmmm, any hint how to do so? Where are these required database settings for KeePassXC-browser stored?

**Comment by varjolintu** [comment]
In the localStorage. One thing you could of course try is to delete the addon folder totally from your Firefox profile and install it again from AMO. There should be a `extensions` directory under your Firefox profile. You can also check if the `storage` folder has anything in it.

**Comment by Dadanaut** [comment]
It has something to do with cookies or stored web databases which get deleted. I disabled my cookie helper programm and now it works...

**Comment by varjolintu** [comment]
What's that cookie helper application you are using? That could be helpful to others who are having the same issue.

**Comment by Dadanaut** [comment]
Ui! You’re right. It was/is „Cookie“ from SweetP Productions (for Mac).

But not the actual version (5.x). Sadly I don’t remember the version I’ve used and deleted. But it was definitively older...

Now I’m using Firefox add-ons for cookies.
